src/cockpit/superuser.py

```diff
@@ -82,13 +82,13 @@ def apply_rule(self, options: Dict[str, object]) -> Optional[Peer]:
             raise RoutingError('access-denied')
 
     # ferny.InteractionResponder
-    async def do_askpass(self, messages: str, prompt: str, hint: str) -> Optional[str]:
```

Comment:
I generally dislike renaming arguments on virtual function implementations.

You also mention yourself in the dialog that the name "error" might indeed be inaccurate.  We could maybe better call this argument `stderr` but then I'd expect to see a corresponding change in ferny.  For the record, I'd also support moving it to the end of the argument list.

Without changing ferny, I think the thing that makes the most sense is to pass the `messages` argument as the last argument to `self.prompt()` and (possibly) adjust the comment at the site where `prompt` is defined.

Reply:
Right, I agree. I reworked it as suggested and added a comment, but not to the declaration (as it's too detached from the place where that "transition" actually happens), but instead to the line below where the prompt signal gets emitted.